src/main/java/io/numaproj/numaflow/reducestreamer/OutputActor.java

```diff
@@ -0,0 +1,87 @@
+package io.numaproj.numaflow.reducestreamer;
+
+import akka.actor.AbstractActor;
+import akka.actor.Props;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Timestamp;
+import io.grpc.stub.StreamObserver;
+import io.numaproj.numaflow.reduce.v1.ReduceOuterClass;
+import io.numaproj.numaflow.reducestreamer.model.Message;
+import io.numaproj.numaflow.reducestreamer.model.Metadata;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Output actor is a wrapper around the gRPC output stream.
+ * It ensures synchronized calls to the responseObserver onNext() and invokes onComplete at the end of the stream.
+ * ALL reduce responses are sent to the response stream actor before getting forwarded to the output gRPC stream.
+ * <p>
+ * More details about gRPC StreamObserver concurrency: https://grpc.github.io/grpc-java/javadoc/io/grpc/stub/StreamObserver.html
+ */
+@Slf4j
+@AllArgsConstructor
+class OutputActor extends AbstractActor {
+    StreamObserver<ReduceOuterClass.ReduceResponse> responseObserver;
+    Metadata md;
+
+    public static Props props(
+            StreamObserver<ReduceOuterClass.ReduceResponse> responseObserver,
+            Metadata md) {
+        return Props.create(OutputActor.class, responseObserver, md);
+    }
+
+    @Override
+    public Receive createReceive() {
+        return receiveBuilder()
+                .match(Message.class, this::sendMessage)
+                .match(ActorResponse.class, this::sendEOF)
+                .build();
+    }
```

Comment:
Can we wrap the message inside ActorResponse so that we will have a common match statement?